server/athenian/api/align/goals/dbaccess.py

```diff
@@ -0,0 +1,71 @@
+"""DB access layer utilities for Align Goals objects."""
+from typing import Sequence
+
+from morcilla import Connection
+import sqlalchemy as sa
+
+from athenian.api.models.state.models import Goal, Team, TeamGoal
+from athenian.api.typing_utils import dataclass
+
+from .exceptions import GoalMutationError
+
+
+@dataclass(frozen=True)
+class GoalCreationInfo:
+    """The information required to create a Goal.
+
+    - the goal itself
+    - the list of TeamGoal-s
+
+    """
+
+    goal: Goal
+    team_goals: Sequence[TeamGoal]
+
+
+async def insert_goal(creation_info: GoalCreationInfo, sdb_conn: Connection) -> int:
+    """Insert the goal and related objects into DB."""
+    await _validate_goal_creation_info(creation_info, sdb_conn)
+    goal_value = creation_info.goal.create_defaults().explode()
+    new_goal_id = await sdb_conn.execute(sa.insert(Goal).values(goal_value))
+    team_goals_values = [
+        {
+            **team_goal.create_defaults().explode(with_primary_keys=True),
+            # goal_id can only be set now that Goal has been inserted
+            "goal_id": new_goal_id,
+        }
+        for team_goal in creation_info.team_goals
+    ]
+    await sdb_conn.execute_many(sa.insert(TeamGoal), team_goals_values)
+    return new_goal_id
+
+
+async def delete_goal(account_id: int, goal_id: int, sdb_conn: Connection) -> None:
+    """Delete a goal from DB with related team goals."""
+    where_clause = sa.and_(Goal.account_id == account_id, Goal.id == goal_id)
+    # no rowcount support in morcilla, no delete ... returning support in sqlalchemy / sqlite,
```

Comment:
In this case we should take a transaction or check that we are inside a transaction.

Reply:
btw if you wish you can add rowcount support to morcilla, why not :+1:

Reply:
added transaction check, caller could execute something else in the same transaction